domain/application/state/application.go

```diff
@@ -658,14 +658,50 @@ SELECT &KeyValue.* FROM model_config WHERE key IN ($S[:])
 	})
 }
 
-// GetStoragePoolByName returns the storage pool with the specified name, returning an error
-// satisfying [storageerrors.PoolNotFoundError] if it doesn't exist.
-func (st *State) GetStoragePoolByName(ctx context.Context, name string) (domainstorage.StoragePool, error) {
+// GetStoragePoolUUID returns the UUID of the storage pool for the specified name.
+// The following errors can be expected:
+// - [storageerrors.PoolNotFoundError] if a pool with the specified name does not exist.
+func (st *State) GetStoragePoolUUID(
+	ctx context.Context,
+	name string,
+) (domainstorage.StoragePoolUUID, error) {
+	db, err := st.DB()
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	var poolUUID domainstorage.StoragePoolUUID
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		poolUUID, err = storagestate.GetStoragePoolUUID(ctx, tx, st, name)
+		return err
+	})
+	if err != nil {
+		return "", errors.Errorf("getting storage pool %q UUID: %w", name, err)
```

Comment:
We don't need to annotate this under our rules. This addition should be done by the caller.